app/assets/javascripts/global-bar-init.js

```diff
@@ -0,0 +1,62 @@
+//= require libs/GlobalBarHelper.js
+
+'use strict'
+window.GOVUK = window.GOVUK || {}
+
+// Bump this if you are releasing a major change to the banner
+// This will reset the view count so all users will see the banner, even if previously seen
+var BANNER_VERSION = 3;
+
+var globalBarInit = {
+  getBannerVersion: function() {
+    return BANNER_VERSION
+  },
+
+  getLatestCookie: function() {
+    return document.cookie.match("(?:^|[ ;])(?:global_bar_seen=)(.+?)(?:(?=;|$))")
+  },
+
+  blacklistedUrl: function() {
+    var paths = ["/register-to-vote", "/done", "/brexit", "/get-ready-brexit-check"]
+
+    return new RegExp(paths.join("|")).test(window.location.pathname)
+  },
+
+  setBannerCookie: function() {
+    var eighty_four_days = 84*24*60*60*1000
+    var expiryDate = new Date(Date.now() + eighty_four_days).toUTCString()
+    var value = JSON.stringify({count: 0, version: globalBarInit.getBannerVersion()})
+    var cookieString = "global_bar_seen=" + value + "; expires=" + expiryDate + ";"
+
+    document.cookie=cookieString
+  },
+
+  makeBannerVisible: function() {
+    document.documentElement.className = document.documentElement.className.concat(" show-global-bar")
+  },
+
+  init: function() {
+    var currentCookieValue = globalBarInit.getLatestCookie()
+
+    if (!globalBarInit.blacklistedUrl()) {
+      if (currentCookieValue === null) {
```

Comment:
I would just call the function inline here and delete line 39:

```suggestion
      if (globalBarInit.getLatestCookie() === null) {
```

And I would do so in the conditionals on line 48 and 54. It's slightly less efficient than caching the value in a variable, but easier to read & less can go wrong.